Add documentation for check and fix: Sliver polygons, Small polygons, Fix small polygons (#10333)
qgis/QGIS-Documentation: docs/user_manual/processing_algs/qgis/checkgeometry.rst

**Algorithm ID**: ``native:checkgeometryselfintersection``

.. include:: ../algs_include.rst
  :start-after: **algorithm_code_section**
  :end-before: **end_algorithm_code_section**

.. _qgischeckgeometrysliverpolygon:

Sliver polygons
------------------

Checks sliver polygons in a polygon layer.
The thinness is the ratio between the area of the minimum square
containing the polygon and the area of the polygon itself (a square has a thinness value of 1).
The thinness value is between 1 and +infinity.
If a polygon has an area higher than the maximum area,
it is skipped (a maximum area value of 0 means no area check).
Polygons having a thinness higher than the maximum thinness are errors.

> To fix sliver polygons, use the "Fix small polygons" algorithm.

Reviewer: ```suggestion
To fix sliver polygons, use the :ref:`qgisfixgeometryarea` algorithm.
```
let's be cautious with alg name.

.. figure:: img/check_geometry_sliver_polygon.png
   :align: center

   Reporting errors on features with sliver polygons.

.. seealso:: :ref:`qgisfixgeometryarea`

Parameters
..........

Basic parameters
^^^^^^^^^^^^^^^^

.. list-table::
   :header-rows: 1
   :widths: 20 20 20 40
   :class: longtable

   * - Label
     - Name
     - Type
     - Description
   * - **Input layer**
     - ``INPUT``
     - [vector: polygon]
     - Layer with the geometries to check.
   * - **Unique feature identifier**
     - ``UNIQUE_ID``
     - [tablefield: any]
     - Field storing unique values for feature identification.
   * - **Maximum thinness**
     - ``MAX_THINNESS``
     - [numeric: double]

       Default: 20.0
     - Maximum thinness value of a polygon to be considered valid.
       Any polygon with a thinness value above this threshold is reported as error.
   * - **Maximum area (map units squared)**
     - ``MAX_AREA``
     - [numeric: double]
  
       Default: 0.0
     - Maximum area of a polygon to be checked.
       Any polygon with an area above this threshold is skipped.

>        A value of 0 means no area check.

Reviewer: ```suggestion
       A value of 0 means all polygons are checked.
```

     - [numeric: integer]

       Default: 8
     - Numerical precision of geometric operations, given as an integer n,
       meaning that two vertices less than 10\ :sup:`-n` apart (in map units)
       are considered to be merged.

Outputs
.......

.. list-table::
   :header-rows: 1
   :widths: 20 20 20 40
   :class: longtable

   * - Label
     - Name
     - Type
     - Description
   * - **Sliver polygon errors**
     - ``ERRORS``
     - [vector: point]
     - Output point layer representing the error locations and information.
       The output layer contains the following fields:

       - ``gc_layerid``: the ID of the input layer.
       - ``gc_layername``: the name of the input layer.
       - ``gc_partidx``
       - ``gc_ringidx``
       - ``gc_vertidx``
       - ``gc_errorx``: the x coordinate of the centroid of the sliver polygon feature.
       - ``gc_errory``: the y coordinate of the centroid of the sliver polygon feature.
       - ``gc_error``: the thinness value of the sliver polygon feature.
       - ``UNIQUE_ID`` field: the unique ID of the input feature that is a sliver polygon.
   * - **Sliver polygon features**
     - ``OUTPUT``
     - [vector: polygon]

>      - Output layer with features containing the sliver polygon features.

Reviewer: ```suggestion
     - Output layer with features containing the sliver polygons.
```

**Algorithm ID**: ``native:checkgeometryangle``

.. include:: ../algs_include.rst
  :start-after: **algorithm_code_section**
  :end-before: **end_algorithm_code_section**

.. _qgischeckgeometryarea:

Small polygons
------------------

> Checks the areas of polygon geometries.

Reviewer: ```suggestion
Detects polygon features whose area is below a specified value as errors.
```

Areas below the area threshold are errors.

.. figure:: img/check_geometry_area.png
   :align: center

   Reporting errors on features with small polygons (below given area threshold).

.. seealso:: :ref:`qgisfixgeometryarea`

Parameters
..........

Basic parameters
^^^^^^^^^^^^^^^^

.. list-table::
   :header-rows: 1
   :widths: 20 20 20 40
   :class: longtable

   * - Label
     - Name
     - Type
     - Description
   * - **Input layer**
     - ``INPUT``
     - [vector: polygon]
     - Layer with the geometries to check.
   * - **Unique feature identifier**
     - ``UNIQUE_ID``
     - [tablefield: any]
     - Field storing unique values for feature identification.
   * - **Area threshold**
     - ``AREATHRESHOLD``
     - [numeric: double]

       Default: 0

>      - Minimum area of a polygon to be considered valid, in squared degrees.

Reviewer: Squared degrees 🤔

Author: I just noticed that, will edit it
